### ISSUE TITLE
use buffer object as source

### ISSUE THREAD

**Issue by dkdl** [alemangui/pizzicato#33]
Hi there,

thanks for the great plugin! The best for playing with effects :) I discovered it recently and now I try to implement it at a website. But at first load I preload multiple tracks with the preloader script from HTML5Rocks. https://www.html5rocks.com/en/tutorials/webaudio/intro/js/buffer-loader.js

And now I'm not sure how to use these buffer objects instead of an url at init. I tried to edit the initializeWithUrl function but no succes.

**Comment by JohnPaulHarold** [comment]
you shouldn't need use that buffer-loader script, just load the sounds you want via Pizzicato. I'm assuming you're loading wavs or mp3s or something. See https://github.com/alemangui/pizzicato#sounds-from-a-file

**Comment by alemangui** [comment]
Hey @dkdl, thanks for the comments :)  @JohnPaulHarold is right, you can just load the file directly with Pizzicato. Is there any reason why you would prefer to use the buffer?

**Comment by dkdl** [comment]
Hello, thank you the replies. I want it to preload, because I play/start 8 files at the same time. Now I'm getting the this.getRawSourceNode error when I want to play the file (too early). 

Before pizzicato I used the bufferloader with promises and then init the sounds. 

But now I have to init, use the callback function, check which files has loaded, and then start the sounds. Is there a way to combine all the sounds instead of checking multiple callbacks?

    var p = new promise.Promise();
    var bufferCount = 0;    
    
    for (var i = 1; i < 5; i++) {

      var tracks['track' + i] = new Pizzicato.Sound({
        source: 'file',
        options: {
          path: 'http://dsfsfsfd.mp3',
          loop: true,
          volume: 1.0
        }
      },
      function() {
        console.log(bufferCount + " A");
        if (bufferCount >= 4 ) {
          p.done();
        } else {
          bufferCount = bufferCount + 1;
        }
      });

    }
    
    return p;

**Comment by JohnPaulHarold** [comment]
I've had to do similar in the past. When I did, it was something like [this]. (https://gist.github.com/JohnPaulHarold/9b3deea677e9e28547b496167ec191e9).

It could probably be done with Promises also. Looking at your code above, I'm not sure if you're using some bespoke Promises polyfill, but with regular Promises, something like [this ](https://gist.github.com/JohnPaulHarold/f2c8a101b5071b9ee5896a8e92c2a893) should work.
